src/controllers/teamController.js

```diff
@@ -43,27 +44,13 @@ const teamcontroller = function (Team) {
     team.createdDatetime = Date.now();
     team.modifiedDatetime = Date.now();
 
-    // Check if a team with the same name already exists
-    Team.findOne({ teamName: team.teamName })
-      .then((existingTeam) => {
-        if (existingTeam) {
-          // If a team with the same name exists, return an error
-          res.status(400).send({ error: 'A team with this name already exists' });
-        } else {
-          // If no team with the same name exists, save the new team
-          team
-            .save()
-            .then((results) => res.send(results).status(200))
-            .catch((error) => {
-              Logger.logException(error, null, `teamName: ${req.body.teamName}`);
-              res.send(error).status(404);
-            });
-        }
-      })
-      .catch((error) => {
+  try {
+        const result = await team.save();
+        res.status(200).send(result);
+    } catch (error) {
         Logger.logException(error, null, `teamName: ${req.body.teamName}`);
-        res.send(error).status(404);
-      });
+        res.status(500).send({ error: 'Internal server error' });
```

Comment:
'Internal server error' could be used in different controllers. We should create either a constant value for this string, or we can even create a helper function that encapsulates a 500 status code response. We can even go further and create a helper function for each status code and accept an input messsage as an argument for that function. 

This could be done in a separate PR.